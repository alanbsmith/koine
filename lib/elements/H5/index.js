import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import fontWeights from '../../modifiers/fontWeights';

const H5 = styled.h5`
  font-size: 24px;
  margin: 0;
  ${applyStyleModifiers(fontWeights)}
`;

export default H5;
