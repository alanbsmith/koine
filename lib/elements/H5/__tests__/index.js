import React from 'react';
import H5 from '../index';
import renderer from 'react-test-renderer';

import { renderWithTheme } from '__tests__/helpers';

describe('H5 Element', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <H5>Header Element</H5>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies mediumWeight modifier', () => {
    const tree = renderWithTheme(
      <H5 modifiers={['mediumWeight']}>Header Element</H5>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
