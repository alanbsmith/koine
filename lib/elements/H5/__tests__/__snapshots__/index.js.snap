// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H5 Element applies mediumWeight modifier 1`] = `
<h5
  className="sc-bdVaJa dpKQGG"
>
  Header Element
</h5>
`;

exports[`H5 Element renders correctly 1`] = `
<h5
  className="sc-bdVaJa dpKQGG"
>
  Header Element
</h5>
`;
